File: 프로그래머스/lv2/42885. 구명보트/구명보트.py
def solution(people, limit):
    ans = 0
    people.sort()
    
    a = 0
    b = len(people) - 1
    while a <= b:
        if a == b:
            ans += 1
            break
        if people[a] + people[b] <= limit:
            ans += 1
            a += 1
            b -= 1
        else:
            ans += 1
            b -= 1
    
    return ans

File: 프로그래머스/lv2/42885. 구명보트/구명보트.js
function solution(people, limit) {
    let ans = people.length;
    people.sort((a, b) => b - a);
    let left = 0;
    let right = people.length - 1;
    while(left < right) {
        if(people[left] + people[right] <= limit){
            ans -= 1;
            right -= 1;
        }
        left += 1;
    }
    
    return ans;
}